src/iface/ethernet.rs

```diff
@@ -21,7 +25,7 @@ use wire::{TcpPacket, TcpRepr, TcpControl};
 use socket::{Socket, SocketSet, AnySocket};
 #[cfg(feature = "socket-raw")]
 use socket::RawSocket;
-#[cfg(feature = "socket-icmp")]
+#[cfg(all(feature = "socket-icmp", feature = "proto-ipv4"))]
```

Comment:
I don't think ICMP sockets should depend on `proto-ipv4`. Sure, right now, since we don't have ICMPv6 at all, they would do nothing, but they should eventually be available, so let's keep them in.

Reply:
If we remove this that would mean a build without default features and without `proto-ipv4` but with `proto-ipv6` would fail. If you're okay with that, I can simply remove this. Otherwise we need to make a generic `IcmpRepr`/`IcmpPacket`

Reply:
Hmmm, I've looked at making ICMP sockets work with ICMPv6 but it looks involved, so let's disable them on non-IPv4 builds for now.

Reply:
Sounds good. I'll create an issue for this and https://github.com/dlrobertson/smoltcp/blob/proto-ipv4/src/lib.rs#L4-L5